paymentsheet/src/main/java/com/stripe/android/paymentelement/confirmation/epms/ExternalPaymentMethodConfirmationDefinition.kt

```diff
@@ -0,0 +1,116 @@
+package com.stripe.android.paymentelement.confirmation.epms
+
+import androidx.activity.result.ActivityResultCaller
+import androidx.activity.result.ActivityResultLauncher
+import com.stripe.android.common.exception.stripeErrorMessage
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.paymentelement.confirmation.ConfirmationDefinition
+import com.stripe.android.paymentelement.confirmation.ConfirmationHandler
+import com.stripe.android.paymentelement.confirmation.intent.DeferredIntentConfirmationType
+import com.stripe.android.payments.core.analytics.ErrorReporter
+import com.stripe.android.payments.paymentlauncher.PaymentResult
+import com.stripe.android.paymentsheet.ExternalPaymentMethodConfirmHandler
+import com.stripe.android.paymentsheet.ExternalPaymentMethodContract
+import com.stripe.android.paymentsheet.ExternalPaymentMethodInput
+import java.lang.IllegalStateException
+import javax.inject.Provider
+
+internal class ExternalPaymentMethodConfirmationDefinition(
+    private val externalPaymentMethodConfirmHandlerProvider: Provider<ExternalPaymentMethodConfirmHandler?>,
```

Comment:
I made this a provider since this handler is updated when the activity/fragment is created and destroyed. Ideally I think the definition should not have to know this but this requires a larger change to `ExternalPaymentMethodInterceptor` to work more similarly to how `IntentConfirmationInterceptor` works (have an interface, implement `DefaultConfirmationInterceptor` with handler inside, external users only are aware of the interface and method)

Reply:
I'll make this update a follow-up task after the overall architecture changes to improve the EPM internal API.